command-line-interface.md

```diff
@@ -0,0 +1,234 @@
+If the runtime is invoked with an unrecognized command, it MUST exit with a nonzero exit code and MAY log a warning to stderr.
+Beyond the above rules, the behavior of the runtime in the presence of commands and options not specified in this document is unspecified.
+
+## Character encodings
+
+This API specification does not cover character encodings, but runtimes SHOULD conform to their native operating system.
+For example, POSIX systems define [`LANG` and related environment variables][posix-lang] for [declaring][posix-locale-encoding] [locale-specific character encodings][posix-encoding], so a runtime in an `en_US.UTF-8` locale SHOULD write its [state](#state) to stdout in [UTF-8][].
+
+## Commands
+
+### create
+
+[Create][create] a container from a [bundle directory][bundle].
+
+* *Arguments*
+    * *`<ID>`* Set the container ID to create.
+* *Options*
+    * *`--bundle <PATH>`* Override the path to the [bundle directory][bundle] (defaults to the current working directory).
+    * *`--pid-file <PATH>`* The runtime MUST write the container PID to this path.
+    * *`--console-socket <PATH>`* The runtime MUST pass the [pseudoterminal master][posix_openpt.3] through the socket at `<PATH>`; the protocol is [described below](#console-socket).
+      `--console-socket` MUST be set when [`process.terminal`][process] is true and MUST NOT be set when [`process.terminal`][process] is not true.
+* *Standard streams:*
+    * If [`process.terminal`][process] is true:
+        * *stdin:* The runtime MUST NOT attempt to read from its stdin.
+        * *stdout:* The handling of stdout is unspecified.
+        * *stderr:* The runtime MAY print diagnostic messages to stderr, and the format for those lines is not specified in this document.
+    * If [`process.terminal`][process] is not true:
+        * *stdin:* The runtime MUST attach its stdin directly to the container process without reading from it.
+        * *stdout:* The runtime MUST attach its stdout directly to the container process without writing to it.
+        * *stderr:* The runtime MUST attach its stderr to the container process, and MUST not write to it unless it exits with a non-zero code.
+* *Environment variables*
+    * *`LISTEN_FDS`:* The number of file descriptors passed.
+      For example, `LISTEN_FDS=2` would mean that the runtime MUST pass file descriptors 3 and 4 to the container process (in addition to the standard streams) to support [socket activation][systemd-listen-fds].
+* *Exit code:* Zero if the container was successfully created and non-zero on errors.
+
+Callers MAY block on this command's successful exit to trigger post-create activity.
+
+#### Console socket
+
+The [`AF_UNIX`][unix-socket] [`SOCK_SEQPACKET`][socket-types] used by [`--console-socket`](#create) handles request and response messages between a client and server.
```

Comment:
This is the first place we introduce a socket listener, "server" is the usual term for such listeners, and "client" is the usual term for processes that connect to such servers.  I'd be ok using "server" and "runtime" though; would that address your concern?

`SOCK_SEQPACKET` is because, as I understand it, the client and server have to agree on socket type without negotiation.  So we have to require something, and `SOCK_SEQPACKET` seems like a better fit.  But maybe I'm wrong and you can mix and match client/server types?